Assets/VRTK/Scripts/VRTK_AdaptiveQuality.cs

```diff
@@ -0,0 +1,708 @@
+        /// </remarks>
+        public float currentRenderScale
+        {
+            get { return VRSettings.renderScale * VRSettings.renderViewportScale; }
+        }
+
+        /// <summary>
+        /// The recommended render target resolution provided by the current `VRDevice`.
+        /// </summary>
+        public Vector2 defaultRenderTargetResolution
+        {
+            get { return RenderTargetResolutionForRenderScale(allRenderScales[defaultRenderScaleLevel]); }
+        }
+
+        /// <summary>
+        /// The current render target resolution.
+        /// </summary>
+        public Vector2 currentRenderTargetResolution
+        {
+            get { return RenderTargetResolutionForRenderScale(currentRenderScale); }
+        }
+
+        #endregion
+
+        #region Private fields
+
+        /// <summary>
+        /// The frame duration in milliseconds to fallback to if the current `VRDevice` specifies no refresh rate.
+        /// </summary>
+        private const float DefaultFrameDurationInMilliseconds = 1000f / 90f;
+
+        private readonly List<float> allRenderScales = new List<float>();
+        private int defaultRenderScaleLevel;
+        private int currentRenderScaleLevel;
+        private float previousMinimumRenderScale;
+        private float previousMaximumRenderScale;
+        private float previousRenderScaleFillRateStepSizeInPercent;
+
+        private int lastRenderScaleChangeFrameCount;
+        private readonly float[] frameTimeRingBuffer = { 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f,
+                                                         0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f };
+        private int frameTimeRingBufferIndex;
+        private bool interleavedReprojectionEnabled;
+        private bool hmdDisplayIsOnDesktop;
+        private float singleFrameDurationInMilliseconds;
+
+        private GameObject debugVisualizationQuad;
+        private Material debugVisualizationQuadMaterial;
+
+        #endregion
+
+        public VRTK_AdaptiveQuality()
+        {
+            renderScales = allRenderScales.AsReadOnly();
+        }
+
+        /// <summary>
+        /// Calculates and returns the render target resolution for a given render scale.
+        /// </summary>
+        /// <param name="renderScale">
+        /// The render scale to calculate the render target resolution with.
+        /// </param>
+        /// <returns>
+        /// The render target resolution for <paramref name="renderScale"/>.
+        /// </returns>
+        public static Vector2 RenderTargetResolutionForRenderScale(float renderScale)
+        {
+            return new Vector2((int)(VRSettings.eyeTextureWidth / VRSettings.renderScale * renderScale),
+                               (int)(VRSettings.eyeTextureHeight / VRSettings.renderScale * renderScale));
+        }
+
+        /// <summary>
+        /// Calculates and returns the biggest allowed maximum render scale to be used for <see cref="maximumRenderScale"/>
```

Comment:
remove the `<see cref` as it will just get ignored, just back tick the word like \`maximumRenderScale\`